File: artemis-entity-tracker/src/net/namekdev/entity_tracker/utils/tuple/Tuple2.java
package net.namekdev.entity_tracker.utils.tuple;

public final class Tuple2<E1, E2> {
	public E1 item1;
	public E2 item2;

	public Tuple2(E1 item1, E2 item2) {
		this.item1 = item1;
		this.item2 = item2;
	}

	public static <E1, E2> Tuple2<E1, E2> create(E1 item1, E2 item2) {
		return new Tuple2<E1, E2>(item1, item2);
	}
}

File: src/commonMain/kotlin/net/namekdev/entity_tracker/utils/tuple/Tuple2.kt
package net.namekdev.entity_tracker.utils.tuple

class Tuple2<E1, E2>(var item1: E1, var item2: E2) {
    companion object {

        fun <E1, E2> create(item1: E1, item2: E2): Tuple2<E1, E2> {
            return Tuple2(item1, item2)
        }
    }
}
